cmd/reject/main.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"log/slog"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/canonical/specs-v2.canonical.com/config"
+	"github.com/canonical/specs-v2.canonical.com/db"
+	"github.com/canonical/specs-v2.canonical.com/google"
+	"github.com/canonical/specs-v2.canonical.com/specs"
+)
+
+// main implements a command that runs as a daemon to auto-reject specs based
+// on how long they have been open. In addition to the daemon mode, it also
+// supports rejecting an individual spec, or performing a dry-run on all specs.
+func main() {
+	var (
+		dryRun      bool
+		googleDocID string
+	)
+	flag.BoolVar(&dryRun, "dry-run", false, "perform a dry run without making actual changes")
+	flag.StringVar(&googleDocID, "google-doc-id", "", "reject a single doc (optional)")
+	flag.Parse()
+
+	cfg := config.MustLoadConfig()
+	logger := config.SetupLogger()
+
+	// Signal handling
+	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer cancel()
+
+	rejectService := initRejectService(cfg, logger)
+	rejectService.Config = specs.RejectConfig{
+		DryRun: dryRun,
+	}
```

Comment:
This should be moved to `initRejectService` (pass `dryRun` to it). Modifying a struct like this after initialization by another function is usually a bad idea.